rclpy/rclpy/task.py

```diff
@@ -61,7 +61,7 @@ def __del__(self) -> None:
 
     def __await__(self) -> Generator[None, None, Optional[T]]:
         # Yield if the task is not finished
-        while not self._done:
+        while not self._done and not self._cancelled:
```

Comment:
Asyncio tasks work a little differently, they have 3 possible states and are considered done if they are not pending:
https://github.com/python/cpython/blob/ee0746af7d7cfc6cc25441726034e4fea4bcf7e5/Lib/asyncio/base_futures.py#L7-L10
https://github.com/python/cpython/blob/403410fa1be036214efa7955127911e5592910db/Lib/asyncio/futures.py#L177-L183
@fujitatomoya what do you think?